### DISCUSSION TITLE
How can I both let VSCode and docsify support alert/notice block rendering?

### DISCUSSION THREAD

**Discussion by pbjacob** [docsifyjs/docsify#1894] [Q&A] [answered]
I want to write an admonition block in my markdown file, and I want it to be well rendered in both VSCode markdown preview window and on docsify webpage. But till now, I can only make either one sucessful, while the other one fails to render. 

For example, by using `!!! info` which is known as admonition syntax, it can be correctly rendered in VSCode because this syntax is already supported by markdown (extended), but docsify won't be able to recgonize it. By using `?>` or `!>`, of course docsify can render them well as shown in docsify's homepage, but it can't be rendered in VSCode preview window. It seams that `?>` or `!>` are not standard markdown syntaxes.

So how can I render an admonition block in both ways under the same syntax?
Thank you very much for any replies.

**Comment by Koooooo-7** [comment, marked as answer]
Hi @pbjacob . I think the workaround is converting the syntax gap by `docsify hook`. i.e.
```js
            hook.beforeEach((text)=> text.replace(/!!!\s*(.*)/g, '!> $1 '))
```

**Reply by pbjacob** [reply]
Thanks a lot! Your post has solved my problem.

After adding the following code in _index.html_, I get what I want.
```
plugins: [
          function transAlertBlock(hook, vm) {
            hook.beforeEach( function (text) {
              return text.replace(/!!! warning\s*(.*)/g, '!> $1 ').replace(/!!! note\s*(.*)/g, '?> $1 ');
            });
          },
        ],
```

**Reply by Koooooo-7** [reply]
no prob. 👌
